vllm_spyre/v1/core/scheduler.py

```diff
@@ -237,5 +242,64 @@ def can_schedule(self, request) -> bool:
             (self.tkv - request.num_prompt_tokens) / self.block_size)
         num_blocks_required -= num_fully_padded_blocks
         cond5 = num_blocks_required <= self.n_free_blocks
+        # check that batch size x tkv is smaller than the max supported number
+        cond6 = self.check_batch_tkv_limit(request)
+
         return start_new_batch or (cond1 and cond2 and cond3 and cond4
-                                   and cond5)
+                                   and cond5 and cond6)
```

Comment:
Food for thought:

First of all, I think later we should rename these `cond*` for something more semantic, while it works, I think with short good names we will have a nice summary of what is needed to satisfy the condition as a whole.

Based on the above comment, the final code would be something better than the suggestion below. I see that `check_batch_tkv_limit` is a more complex function than the other condition, there are sort, and some iterations over the requests. So, if you move this code like the suggestion we would use the short-circuit and only runs this function if the other conditions hold, saving some computation.

```suggestion
                                   and cond5 and self.check_batch_tkv_limit())
```

Another follow up of this suggestion is make sure the function is a _pure function_, that is, without side effects that could mutate `self` for example, otherwise we would have weird behaviors if the logic we write would need always this function to be executed.

That is it, in a nutshell this is an exercise that we would do to think about code readability|simplicity vs performance.

Reply:
great ideas @wallashss ! There is going to be a follow up PR with more runtime optimizations, such as caching the function calls, as to save even more computations if the decode batch did not change in the meantime. Will definitely make it a pure function then. I will also have more 'early exits' as you suggest it. 
We are trying to get a working version in asap for the compiler team to work with the env var. @joerunde